care/facility/api/viewsets/facility_users.py

```diff
@@ -21,13 +25,17 @@ class Meta:
 class FacilityUserViewSet(GenericViewSet, mixins.ListModelMixin):
     serializer_class = UserAssignedSerializer
     filterset_class = UserFilter
-    queryset = User.objects.all()
+    queryset = User.objects.all().prefetch_related(
+        Prefetch("skills", queryset=Skill.objects.filter(userskill__deleted=False))
+    )
     permission_classes = [IsAuthenticated]
     filter_backends = [filters.DjangoFilterBackend]
 
     def get_queryset(self):
         try:
-            facility = Facility.objects.get(external_id=self.kwargs.get("facility_external_id"))
+            facility = Facility.objects.get(
```

Comment:
This will override the queryset property, so the optimization that you make won't work
I would recommend using reverse fk lookup to filter users of a facility

Reply:
oops 😬 Got it!